deployment/ccip/changeset/deployer_group.go

```diff
@@ -103,13 +112,25 @@ func (d *DeployerGroup) GetDeployer(chain uint64) (*bind.TransactOpts, error) {
 }
 
 func (d *DeployerGroup) Enact(deploymentDescription string) (deployment.ChangesetOutput, error) {
+	defer d.reset()
+
```

Comment:
Does this make sense even when mcms is not enabled?

Reply:
Also this object is becoming pretty complex and a critical component since its used by a bunch of changesets, can we have some coverage of this multi-Enact logic?

Or enforce a single-Enact behavior if thats safer / makes more sense.

Reply:
I wonder if a pattern like this would be more obvious/safe for multi proposal. Move the reason and mcmsOpCount to a DeployerGroupContext object.

Then we call it like that
```
d := NewDeployerGroup(...., DeployerGroupContext("proposal description"))
d.DoSomethingOnChain()
// Next Proposal
d = d.WithNewContext("proposal2 description") // Internally clone the DeployerGroupContext and increment the offsets
d.DoSomethingElseOnChain()

d.Enact() // Create the 2 proposals
```